asset-manager.php

```diff
@@ -20,6 +20,19 @@
  */
 defined( 'AM_BASE_DIR' ) || define( 'AM_BASE_DIR', dirname( __FILE__ ) );
 
+if ( ! function_exists( 'am_validate_path' ) ) {
+	/**
+	 * Helper function to validate a path before doing something with it.
+	 *
+	 * @param string $path The path to validate.
+	 *
+	 * @return bool True if the path is valid, false otherwise.
+	 */
+	function am_validate_path( string $path ) : bool {
+		return in_array( validate_file( $path ), [ 0, 2 ], true ) && file_exists( $path );
```

Comment:
Should the `file_exists` check run first to check for its existence before the validation?

Reply:
Nope, this is the preferred order. `validate_file` looks at the path itself without examining the file to see if there are problems with the path (like directory traversal). If the path itself looks safe, then it'll check if the file exists. This is a little more performant than the other way around.

Reply:
TIL